bin/verify-binary-architecture.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+set -xeuo pipefail
+
+if [ $# -ne 2 ]; then
+  echo "Usage: $0 <target-triple> <binary-path>"
+  exit 1
+fi
+TARGET_TRIPLE="$1"
+BINARY_PATH="$2"
+
+# Parse target architecture
+case "${TARGET_TRIPLE}" in
+  "x86_64"*) TARGET_ARCHITECTURE="x86_64" ;;
+  "aarch64"*) TARGET_ARCHITECTURE="aarch64" ;;
+  *) echo "Unknown target architecture '${TARGET_TRIPLE}'"; exit 1 ;;
+esac
+
+# Parse binary architecture and map expected binary architecture for target OS
+case "${TARGET_TRIPLE}" in
+  *"darwin"*)
+    # Parse binary architecture
+    file_output=$(file -b "${BINARY_PATH}")
+    BINARY_ARCHITECTURE=$(echo "${file_output}" | grep -o "x86_64\|arm64" || echo "")
+    # Map expected binary architecture
+    case "${TARGET_ARCHITECTURE}" in
```

Comment:
Let's pass the expected substring in from the workflow rather than having mappings from the target triple to these strings here.

Reply:
The expected substring is different for each target architecture and OS.

In my initial version I had it mapped like so:

```sh
declare -A ARCHITECTURE_PATTERNS=(
  ["darwin"]="x86_64\|arm64"
  ["linux"]="x86-64\|aarch64"
)
declare -A ARCHITECTURE_MAP=(
  ["darwin:x86_64"]="x86_64"
  ["darwin:aarch64"]="arm64"
  ["linux:x86_64"]="x86-64"
  ["linux:aarch64"]="aarch64"
  ["windows:x86_64"]="X64"
  ["windows:aarch64"]="Arm64"
)
```

Which reduced the logic to just:

```sh
BINARY_ARCHITECTURE=$(file "${BINARY_PATH}" | grep -o "${ARCHITECTURE_PATTERNS["linux"]}" || echo "")
EXPECTED_BINARY_ARCHITECTURE="${ARCHITECTURE_MAP["linux:${TARGET_ARCHITECTURE}"]:-}"
```

Way more clear – it sadly doesn't run on `darwin`.

Reply:
Hope its more clear now in the refactored version of the script?